docs/asciidoc-writers-guide.adoc

```diff
@@ -366,6 +366,8 @@ include::{includedir}/xref.adoc[]
 
 include::{includedir}/xref-relative.adoc[]
 
+TIP: The xref inline macro by default resolves the link to HTML, to create a relative link to an asciidoc file in another subdirectory include `ifdef::env-github[:outfilesuffix: .adoc]` below the document header and drop the ".adoc" extension after the filename.
```

Comment:
@ecki, Thanks for pointing out, it shouldn't indeed be limited to just GitHub. 
I am not aware of any generic way, will explore and update the PR.